fvm/transactionStorageLimiter.go

```diff
@@ -21,10 +21,10 @@ func getStorageCapacityFuncFactory(
 	vm *VirtualMachine,
 	ctx Context,
 	_ *TransactionProcedure,
-	st *state.State,
+	sth *state.StateHolder,
 	programs *Programs,
 ) (func(address common.Address) (value uint64, err error), error) {
```

Comment:
how come the storage limiter (and transaction fee deductor) doesn't have a block like this:
```go
	parentState := sth.State()
	childState := sth.NewChild()
	defer func() {
		if mergeError := parentState.MergeState(childState); mergeError != nil {
			panic(mergeError)
		}
		sth.SetActiveState(parentState)
	}()
```

Reply:
I think the storage limiter requires a more complex logic, if storage fails we need to even drop the delta of the previous step. so I didn't add it for now.

Reply:
Good point. I actually have plans for that! I will show you on Monday.